### PR TITLE
TICK-0004 created DAOS interfaces with class.

### PR DESCRIPTION
**Pull request by DanMedl**
TICK-0004 created DAOS interfaces with class.

### CODE REVIEW

**Review by Shaff3004 — CHANGES_REQUESTED** [review]

File: com.atb/src/main/java/daos/CategoryDao.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package daos;
+
+import com.atb.dtos.Category;
+
+import java.util.List;
+
+public interface CategoryDao {
+
+    List<Category> getAllCategories();
+    Category getCategoryByCode(String code);
```
[the remark this answers is not in the document]
Comment (DanMedl) [reply com.atb/src/main/java/daos/CategoryDao.java]: done

File: com.atb/src/main/java/daos/DefaultCategoryDao.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import com.atb.dtos.Category;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DefaultCategoryDao implements CategoryDao {
+
+    private Map<String, Category> categoryStorage = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (DanMedl) [reply com.atb/src/main/java/daos/DefaultCategoryDao.java]: done

File: com.atb/src/main/java/daos/DefaultCategoryDao.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    @Override
+    public List<Category> getAllCategories() {
+        return new ArrayList<>(categoryStorage.values());
+    }
```
[the remark this answers is not in the document]
Comment (DanMedl) [reply com.atb/src/main/java/daos/DefaultCategoryDao.java]: done

File: com.atb/src/main/java/daos/DefaultCategoryDao.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    public Category getCategoryByCode(String code) {
+        return categoryStorage.get(code);
+    }
+    @Override
+    public void addCategory(Category category) {
+        categoryStorage.put(category.getCode(), category);
+    }
+    @Override
+    public void updateCategory(Category category) {
+        if (categoryStorage.containsKey(category.getCode())) {
```
[the remark this answers is not in the document]
Comment (DanMedl) [reply com.atb/src/main/java/daos/DefaultCategoryDao.java]: done

File: com.atb/src/main/java/daos/DefaultCategoryDao.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            categoryStorage.put(category.getCode(), category);
+        } else {
+            System.out.println("Category not found");
+        }
+    }
+    @Override
+    public void removeCategory(Category category) {
+        categoryStorage.remove(category.getCode());
+
```
[the remark this answers is not in the document]
Comment (DanMedl) [reply com.atb/src/main/java/daos/DefaultCategoryDao.java]: done

File: com.atb/src/main/java/daos/DefaultProductDao.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
+package daos;
```
[the remark this answers is not in the document]
Comment (DanMedl) [reply com.atb/src/main/java/daos/DefaultProductDao.java]: done

File: com.atb/src/main/java/daos/DefaultProductDao.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+public class DefaultProductDao implements ProductDao {
+    private Map<String, Product> productStorage = new HashMap<>();
+    @Override
+    public List<Product> getAllProducts() {
+        return new ArrayList<>(productStorage.values());
+    }
+    @Override
+    public Product getProductById(int id) {
+        return productStorage.get(id);
```
[the remark this answers is not in the document]
Comment (DanMedl) [reply com.atb/src/main/java/daos/DefaultProductDao.java]: I delete this method but I had to delete field in ProductDao class. Is it correct? 
All method should be override when we implement this interface and it just couldn’t went through IDEA complier. 
Please advise

File: com.atb/src/main/java/com/atb/daos/DefaultCategoryDao.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
+package com.atb.daos;
```
Comment (Shaff3004) [inline com.atb/src/main/java/com/atb/daos/DefaultCategoryDao.java:1]: for implementation classes package should be com.atb.daos.impl

File: com.atb/src/main/java/com/atb/daos/DefaultCategoryDao.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public void updateCategory(Category category) {
+        categoryStorage.put(category.getCode(), category);
+    }
+
+        @Override
+        public void removeCategory (Category category ){
+            categoryStorage.remove(category.getCode());
+
```
Comment (Shaff3004) [inline com.atb/src/main/java/com/atb/daos/DefaultCategoryDao.java:37]: remove empty line

File: com.atb/src/main/java/com/atb/daos/DefaultProductDao.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
+package com.atb.daos;
```
Comment (Shaff3004) [inline com.atb/src/main/java/com/atb/daos/DefaultProductDao.java:1]: the same here: com.atb.daos.impl

File: com.atb/src/main/java/com/atb/daos/DefaultProductDao.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+import com.atb.dtos.Category;
```
Comment (Shaff3004) [inline com.atb/src/main/java/com/atb/daos/DefaultProductDao.java:3]: Are you sure you need that import in this class?

File: com.atb/src/main/java/com/atb/daos/DefaultProductDao.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+import com.atb.dtos.Product;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DefaultProductDao implements ProductDao {
+    private Map<String, Product> productStorage = new HashMap<>();
```
Comment (Shaff3004) [inline com.atb/src/main/java/com/atb/daos/DefaultProductDao.java:12]: put empty line between class and productStorage

File: com.atb/src/main/java/com/atb/daos/DefaultCategoryDao.java (hunk 9 above)
Comment (DanMedl) [reply com.atb/src/main/java/com/atb/daos/DefaultCategoryDao.java:37]: done

File: com.atb/src/main/java/com/atb/daos/DefaultProductDao.java (hunk 11 above)
Comment (DanMedl) [reply com.atb/src/main/java/com/atb/daos/DefaultProductDao.java:3]: done

File: com.atb/src/main/java/com/atb/daos/DefaultProductDao.java (hunk 12 above)
Comment (DanMedl) [reply com.atb/src/main/java/com/atb/daos/DefaultProductDao.java:12]: done

**Pull request merged**